dts/arm/silabs/xg21/xg21.dtsi

```diff
@@ -360,6 +360,15 @@
 			status = "disabled";
 		};
 
+		letimer0: letimer@5a000000 {
+			compatible = "silabs,letimer-pwm";
```

Comment:
I am a bit confused between `timer` and `pwm`. Since the `pinctrl` are associated to this node, I feel this node should be named `pwm@xxxx`.
And maybe `silabs,letimer-pwm` should be `silabs,pwm-letimer`.

Reply:
Took inspiration from stm32 and max32, and introduced a separate dts node for the timer peripheral, making the pwm node a child of the timer. This also sets us up nicely for supporting e.g. the counter API with the same timers in the future, without . As for the binding name, most bindings in the tree have the form <variant>-pwm. It's also how `silabs,eusart-spi` etc are named.

Reply:
If am not wrong, they have two nodes, one for the timer and another for the PWM itself. I feel that either:
  - `letimer` is dedicated to PWM feature and we just have to adjust some names.
  - `letimer` can be used for other purposes (as a clock source or for a counter) and we need to split it in two nodes (and maybe two drivers, but it seems neither ST or MAX32 do that).

What do you think?

Reply:
Oh sorry, I thought you justified your original implementation. I am going to look at your new version.